File: C/BinarySearch/binary_search/binary_search.c
#include "binary_search.h"

#include <string.h>

int binarySearch( char* array[], int size, char* target )
{
    int left = 0;
    int right = size - 1;

    while( left <= right )
    {
        int middle = left + ( right - left ) / 2;
        int guessIndex = strcmp( array[ middle ], target );

        if( guessIndex == 0 )
        {
            return middle;
        }

        if( guessIndex < 0 )
        {
            left = middle + 1;
        }
        else
        {
            right = middle - 1;
        }
    }

    return -1;
}

File: Python/BinarySearch/binary_search.py
def binary_search(array, target):
    start = 0
    end = len(array) - 1

    while start <= end:
        middle = (start + end) // 2
        result = array[middle]
        if result == target:
            return middle
        elif result < target:
            start = middle + 1
        else:
            end = middle - 1

    return None
